packages/material-ui/src/RadioGroup/RadioGroup.js

```diff
@@ -2,92 +2,29 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
-import warning from 'warning';
 import FormGroup from '../FormGroup';
-import { createChainedFunction, find } from '../utils/helpers';
+import SelectableGroup from '@material-ui/lab/SelectableGroup';
 
-class RadioGroup extends React.Component {
-  radios = [];
+function RadioGroup(props) {
+  const { children, defaultValue, name, onChange, value: valueProp, ...other } = props;
+  let { current: firstLoad } = React.useRef(true);
 
-  constructor(props) {
-    super();
-    this.isControlled = props.value != null;
+  let value = valueProp;
 
-    if (!this.isControlled) {
-      this.state = {
-        value: props.defaultValue,
-      };
+  if (firstLoad) {
```

Comment:
Can we keep the same `isControlled` logic? Maybe using a useState with an initial value function?

Reply:
I didn't really want to keep state here. Wouldn't it look something like
```
  const [value, setValue] = React.useState(() => {
    if(valueProp !== null){
      return defaultValue;
    }
    return null;
  });
  
  if(valueProp !== undefined && valueProp !== value ){
    setValue(valueProp);
  }
```

If we changed it?